### PR TITLE
Replace issue templates with new issue forms

### PR DESCRIPTION
**Pull request by NicolasHug**
This PR introduces the new issue forms that replace issue templates.

This is similar to what was done in torchvision https://github.com/pytorch/vision/pull/4299 and torchaudio, you can see the end result here: https://github.com/pytorch/vision/issues/new/choose (click e.g. on the [bug report](https://github.com/pytorch/vision/issues/new?assignees=&labels=&template=bug-report.yml))

The main new thing is that we can enforce some of the fields to be filled, especially for bug reports. It's also a much cleaner GUI for users IMHO, and we can provide better examples and instructions.

There is still a "blank" template available.

I removed the "Questions" form: we say we close these issues anyway. I replaced it with a direct link to https://discuss.pytorch.org. Since we still have a "blank" template, I think this  covers all previous use-cases properly.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+      If relevant, add a minimal example so that we can reproduce the error by running the code. It is very important for the snippet to be as succinct (minimal) as possible, so please take time to trim down any irrelevant code to help us debug efficiently. We are going to copy-paste your code and we expect to get the same result as you did: avoid any external data, and include the relevant imports, etc. For example:
+
+      ```python
+      # All necessary imports at the beginning
+      import torch
+
+      # A succinct reproducing example trimmed down to the essential parts:
+      t = torch.rand(5, 10)  # Note: the bug is here, we should pass requires_grad=True
+      t.sum().backward()
```
Comment (NicolasHug) [inline .github/ISSUE_TEMPLATE/bug-report.yml:23]: I wasn't super inspired writing this
So if you have a better snippet example, feel free to update the PR :)

**Review by seemethere — APPROVED** [review]
